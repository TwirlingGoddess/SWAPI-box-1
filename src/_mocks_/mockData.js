const mockVehicleApiResponse = {
  count: 39,
  next: 'https://swapi.co/api/vehicles/?page=2',
  previous: null,
  results: [
    {
      cargo_capacity: '50000',
      consumables: '2 months',
      cost_in_credits: '150000',
      created: '2014-12-10T15:36:25.724000Z',
      crew: '46',
      edited: '2014-12-22T18:21:15.523587Z',
      films: ['https://swapi.co/api/films/5/', 
        'https://swapi.co/api/films/1/'],
      length: '36.8',
      manufacturer: 'Corellia Mining Corporation',
      max_atmosphering_speed: '30',
      model: 'Digger Crawler',
      name: 'Sand Crawler',
      passengers: '30',
      pilots: [],
      url: 'https://swapi.co/api/vehicles/4/',
      vehicle_class: 'wheeled'
    }
  ]
};

const mockPeopleApiResponse = {
  count: 87,
  next: 'https://swapi.co/api/people/?page=2',
  previous: null,
  results: [
    {
      birth_year: '19BBY',
      created: '2014-12-09T13:50:51.644000Z',
      edited: '2014-12-20T21:17:56.891000Z',
      eye_color: 'blue',
      films: ['https://swapi.co/api/films/2/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/'],
      gender: 'male',
      hair_color: 'blond',
      height: '172',
      homeworld: 'https://swapi.co/api/planets/1/',
      mass: '77',
      name: 'Luke Skywalker',
      skin_color: 'fair',
      species: ['https://swapi.co/api/species/1/'],
      starships: ['https://swapi.co/api/starships/12/',
        'https://swapi.co/api/starships/22/'],
      url: 'https://swapi.co/api/people/1/',
      vehicles: ['https://swapi.co/api/vehicles/14/',
        'https://swapi.co/api/vehicles/30/']
    }
  ]
};

const mockParseData = [
  {
    birth_year: '19BBY',
    created: '2014-12-09T13:50:51.644000Z',
    edited: '2014-12-20T21:17:56.891000Z',
    eye_color: 'blue',
    films: ['https://swapi.co/api/films/2/',
      'https://swapi.co/api/films/6/',
      'https://swapi.co/api/films/3/'],
    gender: 'male',
    hair_color: 'blond',
    height: '172',
    homeworld: 'https://swapi.co/api/planets/1/',
    mass: '77',
    name: 'Luke Skywalker',
    skin_color: 'fair',
    species: ['https://swapi.co/api/species/1/'],
    starships: ['https://swapi.co/api/starships/12/', 
      'https://swapi.co/api/starships/22/'],
    url: 'https://swapi.co/api/people/1/',
    vehicles: ['https://swapi.co/api/vehicles/14/', 
      'https://swapi.co/api/vehicles/30/']
  }
];

const mockPlanetApiResponse = {
  count: 61,
  next: 'https://swapi.co/api/planets/?page=2',
  previous: null,
  results: [
    {
      climate: 'temperate',
      created: '2014-12-10T11:35:48.479000Z',
      diameter: '12500',
      edited: '2014-12-20T20:58:18.420000Z',
      films: ['https://swapi.co/api/films/6/', 
        'https://swapi.co/api/films/1/'],
      gravity: '1 standard',
      name: 'Alderaan',
      orbital_period: '364',
      population: '2000000000',
      residents: ['https://swapi.co/api/people/5/',
        'https://swapi.co/api/people/68/',
        'https://swapi.co/api/people/81/'],
      rotation_period: '24',
      surface_water: '40',
      terrain: 'grasslands, mountains',
      url: 'https://swapi.co/api/planets/2/'
    }
  ]
};

const mockPeopleObject = [{'Homeworld': undefined, 
  'Name': 'Luke Skywalker', 
  'Population': undefined, 
  'Specie': undefined, 'id': 
'people0', 
  'keyList': 'people'
}];

const mockPeopleObject2 = {
  'Homeworld': undefined, 
  'Name': 'Leia', 
  'Population': undefined, 
  'Specie': undefined, 'id': 
'people2', 
  'keyList': 'people'
};

const mockFavorites = [
  {'Homeworld': undefined, 
    'Name': 'Luke Skywalker', 
    'Population': undefined, 
    'Specie': undefined, 'id': 
'people0', 
    'keyList': 'people'
  },
  {
    'Homeworld': undefined, 
    'Name': 'Leia', 
    'Population': undefined, 
    'Specie': undefined, 'id': 
  'people2', 
    'keyList': 'people'
  }
];

const filmCrawlResponse = [{
  crawl: 'Film Crawl',
  title: 'Movie Title',
  date: '1982-09-18',
  opening_crawl: 'There is unrest in the Galactic'},
{
  crawl: 'Film Crawl',
  title: 'Movie Title',
  date: '1982-09-18',
  opening_crawl: 'There is unrest in the Galactic'
}];


const filmCrawl = {
  crawl: 'Film Crawl',
  title: 'Movie Title',
  date: '1982-09-18',
  opening_crawl: 'There is unrest in the Galactic'
};

const vehicleData = [{
  Class: 'sail barge',
  id: 'vehicles0',
  Model: 'Modified Luxury Sail Barge',
  Name: 'Sail barge',
  Passengers: '500',
  keyList: 'vehicles'
}];

const planetData = [{
  Climate: 'temperate',
  id: 'planets0',
  Name: 'Alderaan',
  Residents: 'Alderaan',
  Population: '2000000000',
  Terrain: 'grasslands, mountains',
  keyList: 'planets'
}];

const parsedVehicleData = [
  {
    'name': 'Sail barge',
    'model': 'Modified Luxury Sail Barge',
    'manufacturer': 'Ubrikkian Industries Custom Vehicle Division',
    'cost_in_credits': '285000',
    'length': '30',
    'max_atmosphering_speed': '100',
    'crew': '26',
    'passengers': '500',
    'cargo_capacity': '2000000',
    'consumables': 'Live food tanks',
    'vehicle_class': 'sail barge',
    'pilots': [],
    'films': [
      'https://swapi.co/api/films/3/'
    ],
    'created': '2014-12-18T10:44:14.217000Z',
    'edited': '2014-12-22T18:21:15.807906Z',
    'url': 'https://swapi.co/api/vehicles/24/'
  }
];

const parsedPlanetData = [
  {
    climate: 'temperate',
    created: '2014-12-10T11:35:48.479000Z',
    diameter: '12500',
    edited: '2014-12-20T20:58:18.420000Z',
    films: ['https://swapi.co/api/films/6/', 
      'https://swapi.co/api/films/1/'],
    gravity: '1 standard',
    name: 'Alderaan',
    orbital_period: '364',
    population: '2000000000',
    residents: ['https://swapi.co/api/people/5/',
      'https://swapi.co/api/people/68/',
      'https://swapi.co/api/people/81/'],
    rotation_period: '24',
    surface_water: '40',
    terrain: 'grasslands, mountains',
    url: 'https://swapi.co/api/planets/2/'
  }
];

export { mockPlanetApiResponse, 
  mockVehicleApiResponse, 
  mockPeopleApiResponse, 
  mockPeopleObject, 
  mockPeopleObject2,
  mockFavorites,
  filmCrawl,
  filmCrawlResponse,
  mockParseData,
  vehicleData,
  planetData,
  parsedVehicleData,
  parsedPlanetData
};

