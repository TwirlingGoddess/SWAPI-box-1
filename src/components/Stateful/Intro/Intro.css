.filmText {
  color: yellow;
  font-family: "Nanum Gothic Coding", monospace;
  font-size: 7vh;
  padding: 4vh; }

.crawl {
  background-color: black;
  padding: 9vh;
  width: 40%; }

.introTitle {
  background-color: black;
  color: yellow;
  font-family: "Nanum Gothic Coding", monospace;
  font-size: 5vh;
  height: 17vh;
  margin-top: 160%;
  position: relative;
  text-align: center;
  text-shadow: 1px 1px darkred;
  width: 107%; }

.intro {
  -webkit-animation: intro 6s ease-out 1s;
  animation: intro 6s ease-out 1s;
  color: yellow;
  font-family: "Nanum Gothic Coding", monospace;
  font-size: 11vh;
  font-weight: 400;
  left: 25%;
  margin: auto;
  opacity: 0;
  position: absolute;
  text-shadow: 2px 2px black;
  top: 39%;
  width: 50%;
  z-index: 1; }

@keyframes intro {
  0% {
    opacity: 0; }
  20% {
    opacity: 1; }
  90% {
    opacity: 1; }
  100% {
    opacity: 0; } }

p {
  color: #FFFF82; }

.board {
  -ms-transform-origin: 50% 100%;
  -webkit-box-shadow: 0 0 10px #9ecaed;
  -webkit-transform-origin: 50% 100%;
  -webkit-transform: perspective(300px) rotateX(25deg);
  border-color: #fb0000;
  bottom: 0;
  box-shadow: 0 0 33px #f10303;
  color: red;
  font-family: "Nanum Gothic Coding", monospace;
  font-size: 370%;
  font-weight: bold;
  height: 26em;
  left: 50%;
  line-height: 20vh;
  margin-left: -14em;
  outline: none;
  overflow: hidden;
  position: absolute;
  text-align: justify;
  transform-origin: 50% 100%;
  transform: perspective(364px) rotateX(20deg);
  width: 16em; }

.board:after {
  bottom: 60%;
  content: ' ';
  left: 0;
  position: absolute;
  right: 0;
  top: 0; }

.content {
  animation: scroll 100s linear 5s;
  position: absolute;
  top: 100%; }

.title, .subtitle {
  font-family: "Nanum Gothic Coding", monospace;
  text-align: center; }

@keyframes scroll {
  0% {
    top: 100%; }
  100% {
    top: -170%; } }
