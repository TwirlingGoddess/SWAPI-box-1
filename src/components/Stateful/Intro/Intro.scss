$main-font : 'Nanum Gothic Coding', monospace;
$yellow-font : yellow;
$black-color :  black;


.filmText {
  color: $yellow-font;
  font-family: $main-font;
  font-size: 7vh;
  padding: 4vh;
}

.crawl {
  background-color: $black-color;
  padding: 9vh;
  width: 40%;
}

.introTitle {
  background-color:$black-color;
  color:$yellow-font;
  font-family: $main-font; 
  font-size: 5vh;
  height: 17vh;
  margin-top: 160%;
  position: relative;
  text-align: center;
  text-shadow: 1px 1px darkred;
  width: 107%;
}


.intro {
  -webkit-animation: intro 6s ease-out 1s;
  animation: intro 6s ease-out 1s;
  color: $yellow-font;
  font-family: $main-font;
  font-size: 11vh;
  font-weight: 400;
  left: 25%;
  margin: auto;
  opacity: 0;
  position: absolute;
  text-shadow: 2px 2px rgb(0, 0, 0);
  top: 39%;
  width: 50%;
  z-index: 1;
}

@keyframes intro {
    0% {
        opacity: 0;
    }
    20% {
        opacity: 1;
    }
    90% {
        opacity: 1;
    }
    100% {
        opacity: 0;
    }
}

p {
  color: #FFFF82;
}

.board {
  -ms-transform-origin: 50% 100%;
  -webkit-box-shadow: 0 0 10px #9ecaed;
  -webkit-transform-origin: 50% 100%;
  -webkit-transform: perspective(300px) rotateX(25deg);
  border-color: #fb0000;
  bottom: 0;
  box-shadow: 0 0 33px #f10303;
  color: red;
  font-family: $main-font;
  font-size: 370%;
  font-weight: bold;
  height: 26em;
  left: 50%;
  line-height: 20vh;
  margin-left: -14em;
  outline: none;
  overflow: hidden;
  position: absolute;
  text-align: justify;
  transform-origin: 50% 100%;
  transform: perspective(364px) rotateX(20deg);
  width: 16em;
}

.board:after {
  bottom: 60%;
  content: ' ';
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
}

.content {
  animation: scroll 100s linear 5s;
  position: absolute;
  top: 100%;
}

.title, .subtitle {
  font-family: $main-font;  
  text-align: center;
}

@keyframes scroll {
    0% {
        top: 100%;
    }
    100% {
        top: -170%;
    }
}