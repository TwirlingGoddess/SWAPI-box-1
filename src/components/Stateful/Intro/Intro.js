import './Intro.css'
import React, { Component } from 'react'
import mainAudio from '../../../assets/star-wars-theme-song.mp3'


class Intro extends Component  {
  constructor() {
    super()
    this.state = {
      randomCrawl: '',
      crawlTitle:'',
      crawlDate: ''
    }
  }

  componentDidMount = async () => {
    try {
      const response = await fetch(`https://swapi.co/api/films`)
      const data = await response.json()
      const selectedFilm = await this.randomOpeningCrawl(data.results)
    }
    catch (error) { 
      throw new Error(error.message);
    };
  }
  
  randomOpeningCrawl = (films) => {
    const filmsLength = films.length
    const randomFilm = Math.floor(Math.random() * filmsLength) + 0;
    const randomCrawl = films[randomFilm].opening_crawl;
    const crawlTitle = films[randomFilm].title;
    const crawlDate = films[randomFilm].release_date;
    this.setState({
      randomCrawl,
      crawlTitle,
      crawlDate
    })

  }

  render(){
    return (
      <div>
        <section className="intro">
          A long time ago, in a galaxy far, far away.... 
        </section>
        <div className='crawl board'>
          <div className='content'>
            <h2 className='filmText'>{this.state.randomCrawl}</h2>
          </div>
          <div className='introTitle'>
            {this.state.crawlTitle}, {this.state.crawlDate}
          </div>
        </div>
        <audio className="mainAudio" autoPlay loop>
          <source src={mainAudio} />
        </audio>
      </div>
    )
  }
}


export default Intro