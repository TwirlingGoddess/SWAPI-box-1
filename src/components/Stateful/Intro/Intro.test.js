import React from 'react';
import { shallow, mount } from 'enzyme';
import Intro from './Intro';
import { mockPlanetApiResponse, mockVehicleApiResponse, mockPeopleApiResponse, mockPeopleObject } from '../../../_mocks_/mockData';

/* eslint-env mocha */

describe('Intro', () => {
  let intro;

  beforeEach(() => {
    intro = shallow(<Intro />, { disableLifecycleMethods: true });
    window.fetch = jest.fn()
  });
  it('matches snapshot', () => {
    expect(intro).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    describe('when the status is ok', () => {
      beforeEach(() => {
        window.fetch.mockImplementation(() => {
          return Promise.resolve({ 
            status: 200,
            ok: true,
            json: () => Promise.resolve( mockPeopleApiResponse )
          })
        })
      })

      it('calls fetch with the correct data', () => {
        const url = 'https://swapi.co/api/films';        
        intro.instance().componentDidMount()
        expect(window.fetch).toHaveBeenCalledWith('https://swapi.co/api/films')
      })

      describe('when the status is not ok', () => {

        it('throws an error if fetch fails', async () => {
          expect(renderedComponent.state('errorStatus')).toEqual('Error adding grocery')
          expect(window.fetch).rejects.toEqual('error');
        })
      })

    })
  })


});