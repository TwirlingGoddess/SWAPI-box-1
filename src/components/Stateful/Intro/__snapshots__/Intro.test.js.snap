// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Intro />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="intro"
>
          A long time ago, in a galaxy far, far away....
</section>,
        <div
          className="crawl board"
>
          <div
                    className="content"
          >
                    <h2
                              className="filmText"
                    />
          </div>
          <div
                    className="introTitle"
          >
                    
                    , 
                    
          </div>
</div>,
        <audio
          autoPlay={true}
          className="mainAudio"
          loop={true}
>
          <source
                    src="star-wars-theme-song.mp3"
          />
</audio>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A long time ago, in a galaxy far, far away....",
          "className": "intro",
        },
        "ref": null,
        "rendered": "A long time ago, in a galaxy far, far away....",
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="content"
>
              <h2
                            className="filmText"
              />
</div>,
            <div
              className="introTitle"
>
              
              , 
              
</div>,
          ],
          "className": "crawl board",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2
                className="filmText"
/>,
              "className": "content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "filmText",
              },
              "ref": null,
              "rendered": "",
              "type": "h2",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                ", ",
                "",
              ],
              "className": "introTitle",
            },
            "ref": null,
            "rendered": Array [
              "",
              ", ",
              "",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoPlay": true,
          "children": <source
            src="star-wars-theme-song.mp3"
/>,
          "className": "mainAudio",
          "loop": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "star-wars-theme-song.mp3",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        "type": "audio",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="intro"
>
            A long time ago, in a galaxy far, far away....
</section>,
          <div
            className="crawl board"
>
            <div
                        className="content"
            >
                        <h2
                                    className="filmText"
                        />
            </div>
            <div
                        className="introTitle"
            >
                        
                        , 
                        
            </div>
</div>,
          <audio
            autoPlay={true}
            className="mainAudio"
            loop={true}
>
            <source
                        src="star-wars-theme-song.mp3"
            />
</audio>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A long time ago, in a galaxy far, far away....",
            "className": "intro",
          },
          "ref": null,
          "rendered": "A long time ago, in a galaxy far, far away....",
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="content"
>
                <h2
                                className="filmText"
                />
</div>,
              <div
                className="introTitle"
>
                
                , 
                
</div>,
            ],
            "className": "crawl board",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2
                  className="filmText"
/>,
                "className": "content",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "filmText",
                },
                "ref": null,
                "rendered": "",
                "type": "h2",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  ", ",
                  "",
                ],
                "className": "introTitle",
              },
              "ref": null,
              "rendered": Array [
                "",
                ", ",
                "",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoPlay": true,
            "children": <source
              src="star-wars-theme-song.mp3"
/>,
            "className": "mainAudio",
            "loop": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "star-wars-theme-song.mp3",
            },
            "ref": null,
            "rendered": null,
            "type": "source",
          },
          "type": "audio",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
